### PR TITLE
Update ssl.mode for PG CDC doc

### LINKED ISSUE #2605: Document: feat(cdc): support for verify-ca and verify-full of Postgres SSL
This issue tracks the documentation update needed for the merged PR #18015.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/18015
Source PR Merged At: 2024-09-12T14:37:54Z

If it is a major improvement that deserves a new page or a new section in the documentation, please check if we should label it as an experiment feature.

### PR DESCRIPTION
**Pull request by WanYixian**
## Description

- Add `verify-ca`, and `verify-full` for SSL.mode
- Add `ssl.root.cert`

## Related code PR

https://github.com/risingwavelabs/risingwave/pull/18015

## Related doc issue

Resolve https://github.com/risingwavelabs/risingwave-docs/issues/2605

## Rendered preview

PG:
image

MySQL:
image

## Checklist

- [ ] I have checked the doc site preview, and the updated parts look good.
- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`emile-00`, `hengm3467`, & `WanYixian`).

### CODE REVIEW

File: docs/guides/ingest-from-mysql-cdc.md [hunk 1]
```diff
@@ -168,7 +168,8 @@ All the fields listed below are required. Note that the value of these parameter
 |database.name| Name of the database. Note that RisingWave cannot read data from a built-in MySQL database, such as `mysql`, `sys`, etc.|
 |table.name| Name of the table that you want to ingest data from. |
 |server.id| Required if creating a shared source. A numeric ID of the database client. It must be unique across all database processes that are running in the MySQL cluster. If not specified, RisingWave will generate a random ID.|
-|ssl.mode| Optional. The `ssl.mode` parameter determines the level of SSL/TLS encryption for secure communication with MySQL. It accepts three values: `disabled`, `preferred`, and `required`. The default value is `disabled`. When set to `required`, it enforces TLS for establishing a connection.|
+|ssl.mode| Optional. The `ssl.mode` parameter determines the level of SSL/TLS encryption for secure communication with MySQL. Accepted values are `disabled`, `preferred`,`required`, `verify-ca`, and `verify-full`. The default value is `disabled`. When set to `required`, `verify-ca`, or `verify-full`, it enforces TLS for establishing a connection.|
```
Comment (StrikeW) [inline docs/guides/ingest-from-mysql-cdc.md]: Actually the https://github.com/risingwavelabs/risingwave/pull/18015 is for Postgres source only, so we don't need to change doc of MySQL.

File: docs/guides/ingest-from-mysql-cdc.md [hunk 2]
```diff
@@ -168,7 +168,8 @@ All the fields listed below are required. Note that the value of these parameter
...
+| ssl.root.cert | Optional. Specify the root certificate secret.|
```
Comment (StrikeW) [inline docs/guides/ingest-from-mysql-cdc.md]: ditto

File: docs/guides/ingest-from-postgres-cdc.md [hunk 3]
```diff
@@ -202,7 +202,8 @@ Unless specified otherwise, the fields listed are required. Note that the value
...
-|ssl.mode| Optional. The `ssl.mode` parameter determines the level of SSL/TLS encryption for secure communication with Postgres. It accepts three values: `disabled`, `preferred`, and `required`. The default value is `disabled`. When set to `required`, it enforces TLS for establishing a connection.|
+|ssl.mode| Optional. The `ssl.mode` parameter determines the level of SSL/TLS encryption for secure communication with Postgres. Accepted values are `disabled`, `preferred`, `required`, `verify-ca`, and `verify-full`. The default value is `disabled`. <ul><li>When set to `required`, it enforces TLS for establishing a connection; </li><li>When set to `verify-ca`, it verifies that the server is trustworthy by checking the certificate chain up to the root certificate stored on the client;</li><li>When set to `verify-full`, it verifies the certificate and also ensures the server hostname matches the name in the certificate.</li></ul> |
+| ssl.root.cert | Optional. Specify the root certificate secret.|
```
Comment (StrikeW) [inline docs/guides/ingest-from-postgres-cdc.md]: I think we need to clarify the term `secret`, which is related to the our own secret manager. The user needs to `CREATE SECRET` first before filling in here.

File: docs/guides/ingest-from-mysql-cdc.md (hunk 1 above)
Comment (WanYixian) [reply docs/guides/ingest-from-mysql-cdc.md]: Saw the supported values for ssl.mode in MySQL cdc also changed, so still no need to document this? 🤔 
image

Comment (StrikeW) [reply docs/guides/ingest-from-mysql-cdc.md]: You can scroll up a bit, mysql source will reject those PG specific options. The intention of  code here is to make them a no-op.

**Comment by WanYixian** [comment]
All updated, please take another look @StrikeW

**Review by StrikeW — APPROVED** [review]
LGTM!

**Pull request merged**